makefile

```diff
@@ -0,0 +1,8 @@
+VERSION := $(shell git describe --tags)
+# There are the default architecture values can be changed via 'make OS=linux ARCH=amd64'
+OS = darwin
```

Comment:
So the normal make would not have OS/ARCH? I could also just print them from `uname` for the all build, though this is not exactly the same format used by GOOS/GOARCH.
What's this -w linker business?
I think I get rest.